### PR TITLE
Set dotnet-watch env variable on inner process

### LINKED ISSUE #52: Make it possible to detect when a process is running inside dotnet-watch
When using approvaltests or similar it will pop up a diff tool when the test does not match the 'gold' template.

Is there a way to know if running inside dnx-watch? If not would you consider adding an environmental variable called `DNX_WATCH=1` (or `DOTNET_WATCH=1` for rc2 so you can know if you are running inside DNX watch?

### PR DESCRIPTION
**Pull request by natemcmaster**
Makes it possible to detect when running inside dotnet-watch by setting 'DOTNET_WATCH=1'

Resolves #52

### CODE REVIEW

File: src/Microsoft.DotNet.Watcher.Tools/ProcessSpec.cs [hunk 1]
```diff
@@ -11,6 +11,8 @@ public class ProcessSpec
     {
         public string Executable { get; set; }
         public string WorkingDirectory { get; set; }
+        public IDictionary<string, string> EnvironmentVariables { get; }
+            = new Dictionary<string, string>();
```
Comment (cesarbs) [inline src/Microsoft.DotNet.Watcher.Tools/ProcessSpec.cs]: nit: leave it on the same line

File: test/Microsoft.DotNet.Watcher.Tools.FunctionalTests/TestProjects/KitchenSink/Program.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
```
Comment (cesarbs) [inline test/Microsoft.DotNet.Watcher.Tools.FunctionalTests/TestProjects/KitchenSink/Program.cs]: Add copyright header.

Comment (natemcmaster) [inline test/Microsoft.DotNet.Watcher.Tools.FunctionalTests/TestProjects/KitchenSink/Program.cs]: Done

**Comment by natemcmaster** [comment]
rebased. :watch: on ci passes.

**Pull request merged**